### DISCUSSION TITLE
Best way to reproduce high resolution animations on PIXI

### DISCUSSION THREAD

**Discussion by KreNtal** [pixijs/pixijs#9083] [Q&A]
Does anyone know what's the best way (talking about performance and smoothness) to reproduce a high resolution animation (like 1920x1080 or more) in PIXI?
Is AnimatedSprite a good way or is it better to use Spine or something else?

**Comment by miltoncandelero** [comment]
single frame quality will be determined by the size of your texture.
smoothness of the animation will be dictated by the amount of frames you have. A 60fps spritesheet will be smooth at 60hz screens but will be jumpy at 144hz screens.
Spine will look the smoothest at any hz
(assuming you don't drop fps)

As for performance, the more quality you have, the better hardware I expect you to have 👀

**Reply by KreNtal** [reply]
Yeah, I already have an animation 1920x1080@60FPS that it is not so long also, like 5s.
Now I want to know what's the best way to use it in PIXI.
Should I export it as frame by frame animation and use it as AnimatedSprite,
should I export it as a video and use it as a VideoTexture or
it is better to convert it to a Spine object?

I would like to know if there are other methods and what's the best/correct one.

**Reply by bigtimebuddy** [reply]
If you're doing full screen animation, high-resolution you'd be better to use video. That will give you the best compression options and smooth playback. Spine can definitely work, but only for some types of assets (e.g., character animation with a lot of re-use of parts)

**Reply by KreNtal** [reply]
Ok thanks a lot, think I will go with the video.

**Reply by KreNtal** [reply]
Sorry for annoying you again. I'm on PIXI 6.5.8 (cannot use 7 because I need Spine).
I noticed that VideoBaseTexture was removed in PIXI 6.
So what's the correct way to manage the video inside pixi canvas?

**Reply by bigtimebuddy** [reply]
Also, Spine now is available on v7, that just happened this week.

Video elements can simply be passed to a Texture like this:

```js
const video = document.createElement('video'); // also could be path/to/video.mp4
const texture = Texture.from(video);
```

Here's another example: https://pixijs.io/examples/#/sprite/video.js

**Reply by KreNtal** [reply]
Ok I already found that example but I don't understand how can I play, stop, loop the VideoTexture, because Texture does not have those methods.

**Reply by bigtimebuddy** [reply]
`texture.baseTexture.resource.source` is reference to the HTMLVideoElement. You can control the playback with that.

**Reply by KreNtal** [reply]
I tried and everything works except the video has alpha and it does not properly works for every device. I tried a lots of things like changing format, loading more tha one source in the html video element but it does not works for Mac and iPhone.
Do you have some recommendations on this?

**Reply by SuperSodaSea** [reply]
Not all browsers support all formats (for example, Safari doesn't seem to support transparent WebM video), so you may need to prepare videos in multiple formats for different browsers, for example WebM for Chrome and HEVC for Safari, and choose them carefully when loading.

Here are some related blogs that may help you:

<https://rotato.app/blog/transparent-videos-for-the-web>
<https://css-tricks.com/overlaying-video-with-transparency-while-wrangling-cross-browser-support/>
<https://benfrain.com/how-to-use-videos-with-alpha-transparency-on-the-web/>
